orchestrator/careplancontributor/applaunch/zorgplatform/service.go

```diff
@@ -225,7 +224,159 @@ func (s *Service) registerFhirClientFactory(config Config) {
+	// - Patient is contained in the Task.encounter but separately requested to include the Practitioner
+	// - ServiceRequest is not provided by Zorgplatform, so we need to create one based on the Task and Encounter
+	var task map[string]interface{}
+	if err = fhirClient.ReadWithContext(ctx, "Task/"+launchContext.WorkflowId, &task); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to fetch Task resource (id=%s): %w", launchContext.WorkflowId, err)
+	}
+	// Search for Encounter, contains Encounter, Organization and Patient
+	if task["context"] == nil {
+		return &user.SessionData{}, fmt.Errorf("task.context is not provided")
+	}
+	taskContext := task["context"].(map[string]interface{})
+	encounterId := strings.Split(taskContext["reference"].(string), "/")[1]
+	var encounterSearchResult fhir.Bundle
+	if err = fhirClient.ReadWithContext(ctx, "Encounter", &encounterSearchResult, fhirclient.QueryParam("_id", encounterId)); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to fetch Encounter resource (id=%s): %w", encounterId, err)
+	}
+	var encounter fhir.Encounter
+	if err := coolfhir.ResourceInBundle(&encounterSearchResult, coolfhir.EntryIsOfType("Encounter"), &encounter); err != nil {
+		return &user.SessionData{}, fmt.Errorf("get Encounter from Bundle (id=%s): %w", encounterId, err)
+	}
+	// Get Patient from bundle, specified by Encounter.subject
+	if encounter.Subject == nil || encounter.Subject.Reference == nil {
+		return &user.SessionData{}, fmt.Errorf("encounter.subject does not contain a reference to a Patient")
+	}
+
+	if encounter.ServiceProvider == nil || encounter.ServiceProvider.Reference == nil {
+		return &user.SessionData{}, fmt.Errorf("encounter.serviceProvider does not contain a reference to an Organization")
+	}
+
+	// Get Organization from bundle, specified by Encounter.serviceProvider
+	var organization fhir.Organization
+	if err := coolfhir.ResourceInBundle(&encounterSearchResult, coolfhir.EntryHasID(*encounter.ServiceProvider.Reference), &organization); err != nil {
+		return &user.SessionData{}, fmt.Errorf("get Organization from Bundle (id=%s): %w", encounterId, err)
+	}
+
+	var patientAndPractitionerBundle fhir.Bundle
+	if err = fhirClient.ReadWithContext(ctx, "Patient", &patientAndPractitionerBundle, fhirclient.QueryParam("_include", "Patient:general-practitioner")); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to fetch Patient and Practitioner bundle: %w", err)
+	}
+	var patient fhir.Patient
+	if err := coolfhir.ResourceInBundle(&patientAndPractitionerBundle, coolfhir.EntryHasID(*encounter.Subject.Reference), &patient); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to find Patient resource in Bundle (id=%s): %w", *encounter.Subject.Reference, err)
+	}
+	var practitioner fhir.Practitioner
+	//TODO: The Practitioner has no indetifier set, so we cannot ensure this is the launched Practitioner. Verify with Zorgplatform
```

Comment:
Question is which one we need to give precedence, the practitioner that created the ServiceRequest or the one that created the Task in SCP. @jorritspee ?